internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -171,61 +211,55 @@ func hasCycles(graph *Graph, used map[string]bool, path map[string]bool, currNod
 	return false
 }
 
-func buildDependencyGraph(s convertSidecarOpts) (*Graph, error) {
-	dependencyGraph := Graph{nodes: make(map[string][]string)}
+func buildDependencyGraph(s convertSidecarOpts) (*graph, error) {
+	dependencyGraph := graph{nodes: make(map[string][]string), cycle: make([]string, 0)}
 
-	// sidecar dependencies
+	// Add any sidecar dependencies.
 	for name, sidecar := range s.sidecarConfig {
-		// add any existing dependency to the graph
-		if len(sidecar.DependsOn) != 0 {
-			for dep := range sidecar.DependsOn {
-				// containers being depended on must exist
-				if s.sidecarConfig[dep] == nil && dep != s.workloadName {
-					return nil, errInvalidContainer
-				}
-
-				dependencyGraph.Add(name, dep)
-			}
-		}
-	}
-
-	// add any image dependencies to the graph
-	if len(s.imageConfig.DependsOn) != 0 {
-		for dep := range s.imageConfig.DependsOn {
-			// containers being depended on must exist
+		for dep := range sidecar.DependsOn {
 			if s.sidecarConfig[dep] == nil && dep != s.workloadName {
 				return nil, errInvalidContainer
 			}
 
-			dependencyGraph.Add(s.workloadName, dep)
+			dependencyGraph.Add(name, dep)
+		}
+	}
+
+	// Add any image dependencies.
+	for dep := range s.imageConfig.DependsOn {
+		if s.sidecarConfig[dep] == nil && dep != s.workloadName {
+			return nil, errInvalidContainer
 		}
+
+		dependencyGraph.Add(s.workloadName, dep)
 	}
 
 	return &dependencyGraph, nil
 }
 
-type Graph struct {
+type graph struct {
 	nodes map[string][]string
+	cycle []string
 }
 
-func (graph *Graph) Add(fromNode, toNode string) {
+func (g *graph) Add(fromNode, toNode string) {
```

Comment:
Can we make this function private too, since we aren't reusing this struct elsewhere?

Reply:
Good point! Adjusted.